x-pack/plugins/ml/public/application/datavisualizer/data_drift/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,416 @@
+            : undefined;
+
+        if (Array.isArray(matchingDataViews) && matchingDataViews.length > 0) {
+          const foundDataView = matchingDataViews.find((d) => {
+            return d.timeFieldName === timeFieldName;
+          });
+
+          if (foundDataView) {
+            setFoundDataViewId(foundDataView.id);
+          } else {
+            setDataViewMsg(
+              i18n.translate(
+                'xpack.ml.dataDrift.indexPatternsEditor.hasDataViewWithDifferentTimeField',
+                {
+                  defaultMessage: `Found a data view matching pattern '{indexPattern}' but with a different time field. Creating a new data view to analyze data drift.`,
+                  values: { indexPattern: indicesName },
+                }
+              )
+            );
+          }
+        }
+      }
+    };
+
+    getMatchingDataView();
+
+    return () => {
+      unmounted = true;
+    };
+  }, [referenceIndexPattern, productionIndexPattern, timeField, dataViews]);
+  const createDataViewAndRedirectToDataDriftPage = async (createAdHocDV = false) => {
+    // Create adhoc data view
+    const indicesName = getDefaultIndexPattern(referenceIndexPattern, productionIndexPattern);
+
+    const timeFieldName =
+      Array.isArray(timeField) && timeField.length > 0 ? timeField[0].value : undefined;
+
+    let dataView;
+
+    if (!foundDataViewId) {
+      const defaultDataViewName =
+        dataViewMsg === undefined
+          ? indicesName
+          : `${indicesName}${timeFieldName ? '-' + timeFieldName : ''}`;
+
+      const modifiedDataViewName = dataViewName === '' ? defaultDataViewName : dataViewName;
+      if (canEditDataView && createAdHocDV === false) {
+        dataView = await dataViews.createAndSave({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      } else {
+        dataView = await dataViews.create({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      }
+    }
+    const dataViewId = foundDataViewId ?? dataView?.id;
+    const url = await locator.getUrl({
+      page: ML_PAGES.DATA_DRIFT,
+      pageState: {
+        index: dataViewId,
+        reference: referenceIndexPattern,
+        production: productionIndexPattern,
+        timeFieldName,
+      },
+    });
+
+    await navigateToPath(url);
+  };
+
+  const hasError =
+    refError !== undefined ||
+    comparisonError !== undefined ||
+    !productionIndexPattern ||
+    !referenceIndexPattern;
+
+  const firstSetOfSteps = [
+    {
+      title: 'Pick index pattern for reference data',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'reference'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.referenceData"
+                defaultMessage="Select an index pattern to specify the reference data."
```

Comment:
Do we need a label for the field, as it is basically a repeat of the label on the step? If so, what about just using `Index pattern`?

<img width="497" alt="image" src="https://github.com/elastic/kibana/assets/7405507/3a896e74-c5e5-42dc-810f-b8b2f79efb88">

Any thoughts @szabosteve ?

Reply:
Yeah, the title of the step says it all. I suggested this addition in the draft document. Seeing the text in the UI is a different experience. I think the helper text above the field can be deleted or if it feels empty and/or weird it can be as simple as `Index pattern`.

Reply:
Updated here [`f2f5a38` (#162853)](https://github.com/elastic/kibana/pull/162853/commits/f2f5a385cb03e8aaf0b41ee595c410193686928f)